### DISCUSSION TITLE
Remove all "is caller an account" restrictions

### DISCUSSION THREAD

**Discussion by Stebalien** [filecoin-project/FIPs#497] [Enhancements - Technical]
With the up and coming user programmable actors, we should remove all the remaining "is caller an account (or multisig) restrictions". Specifically:

1. In the payment channel actor, use [FIP0044](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0044.md) instead of signatures and allow any actors to be the to and from.
2. In the multisig actor, allow _any_ actor to be listed as a "signer".
3. In the miner actor, allow _any_ actor to be an owner and/or a control address.
    - NOTE: the restrictions on the _worker_ address cannot be dropped. This will continue to be a BLS account.
4. In the power actor, allow _any_ actor to construct a new miner.
5. In the market actor, allow _any_ actor to call `publish_storage_deals` and `add_balance`.

**Comment by anorth** [comment]
This is a subset of https://github.com/filecoin-project/builtin-actors/issues/424. In general I don't think we'll need a FIP for this, though parts of it might be mentioned in other FIPs, especially...

This is part of #401, as it's only relevant to methods we intend to be called by other actors. I've reviewed the current APIs (https://github.com/filecoin-project/builtin-actors/issues/766) and proposed which methods to export in https://docs.google.com/spreadsheets/d/1McCcCDHLagBFd5miTcTDBPuTIzmY6uKTxX6SS-xsKVY/edit#gid=0. Assuming we take the approach described, we could drop the restrictions as we export those methods.

Some of those you mention I've marked as "Later", as in we could drop the restrictions but I propose we don't rush it.

cc @arajasek @jennijuju

**Reply by Stebalien** [reply]
Ah, awesome!
